espnet2/torch_utils/load_pretrained_model.py

```diff
@@ -0,0 +1,52 @@
+from pathlib import Path
+from typing import Any
+from typing import Union
+
+import torch
+import torch.nn
+import torch.optim
+
+
+def load_pretrained_model(
```

Comment:
I deleted `espnet2.train_utils.checkpoint.py`  and created  `espnet2.train_utils.load_pretrained_model.py`. load_checkpoint and save_checkpoint are removed because the procedures are simplified and such functions seem redundant.

Reply:
is this used for inference as well?
it would be better to unify all model load functions via this?

Reply:
Sorry for not clear explanations. 
For inference, torch.load() is used simply. `load_pretrained_model` is used for training for fine tuning or distillation.

Reply:
`load_pretrained_model`  can apply partially parameters of some parts of the model.
However, to be honest, there are practical example of fine-tuning, so you don't need to care this much. We may need to modify this in the future.

Reply:
ok